src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/CheckinDetailViewModel.swift

```diff
@@ -11,10 +12,115 @@ final class CheckinDetailViewModel {
 		_ checkin: Checkin
 	) {
 		self.checkin = checkin
+		let components: Set<Calendar.Component> = [.minute, .hour]
+		let eventDuration = Calendar.current.dateComponents(
+			components,
+			from: checkin.traceLocationStartDate ?? Date(),
+			to: checkin.traceLocationEndDate ?? Date()
+		)
+		pickerValues = TwoComponentsIntegerPicker(
+			firstComponentSelectedValue: eventDuration.hour,
+			secondComponentSelectedValue: eventDuration.minute,
+			firstComponentValues: Array(0...23),
+			secondComponentValues: [0, 15, 30, 45]
+		)
 	}
+	
+	// MARK: - Internal
+	
+	let numberOfComponents = 4
+	
+	@OpenCombine.Published var descriptionLabelTitle: String?
+	@OpenCombine.Published var addressLabelTitle: String?
 
-	// MARK: - Private
+	func pickerView(viewForRow row: Int, forComponent component: Int) -> UIView {
+		let pickerLabel = ENALabel()
+		let weight: UIFont.Weight = (component == 0 || component == 2) ? .regular : .semibold
+		pickerLabel.font = .enaFont(for: .subheadline, weight: weight, italic: false)
+		pickerLabel.adjustsFontForContentSizeCategory = false
+		
+		switch component {
+		case 0:
+			pickerLabel.text = "\(pickerValues.firstComponentValues[row])"
+			pickerLabel.textAlignment = .center
+		case 1:
+			pickerLabel.text = AppStrings.Checkin.Details.hours
+			pickerLabel.textAlignment = .left
+		case 2:
+			pickerLabel.text = "\(pickerValues.secondComponentValues[row])"
+			pickerLabel.textAlignment = .center
+		case 3:
+			pickerLabel.text = AppStrings.Checkin.Details.minutes
+			pickerLabel.textAlignment = .left
+		default:
+			pickerLabel.text = ""
+		}
+		return pickerLabel
+	}
+	
+	func pickerView(_ pickerView: UIPickerView, widthForComponent component: Int) -> CGFloat {
+		switch component {
+		case 0, 2:
+			return pickerView.bounds.width * 0.15
```

Comment:
Is 0.15 also working for different screensizes?

Reply:
only tried iphone 8 and 12 to be honest, but if it didn't work for small size what should i do? because the picker in the design has less width than the  screen

Reply:
I would just test it on the smallest device. If it works, its fine.